app/services/encryption/multi_region_kms_migration/profile_migrator.rb

```diff
@@ -0,0 +1,58 @@
+module Encryption
+  module MultiRegionKmsMigration
+    class ProfileMigrator
+      attr_reader :profile
+
+      def initialize(profile)
+        @profile = profile
+      end
+
+      def migrate!
+        return if profile.encrypted_pii.blank? || profile.encrypted_pii_recovery.blank?
+        return if profile.encrypted_pii_multi_region.present? ||
+                  profile.encrypted_pii_recovery_multi_region.present?
+
+        encrypted_pii_multi_region = migrate_ciphertext(profile.encrypted_pii)
+        encrypted_pii_recovery_multi_region = migrate_ciphertext(profile.encrypted_pii_recovery)
+        profile.update!(
+          encrypted_pii: profile.encrypted_pii,
+          encrypted_pii_multi_region: encrypted_pii_multi_region,
+          encrypted_pii_recovery: profile.encrypted_pii_recovery,
+          encrypted_pii_recovery_multi_region: encrypted_pii_recovery_multi_region,
+        )
+      end
+
+      private
+
+      def migrate_ciphertext(ciphertext_string)
```

Comment:
would it be useful at all to add NewRelic timing instrumentation around this?

Reply:
Potentially for the `migrate!` method. I think most useful though will be the method that queries for all of the records we eventually operate on.

Reply:
I just added a method tracer for this one. I took a note to make sure we also have one on the class that actually scans for profiles to migrate